### DISCUSSION TITLE
Is it possible to have many product classes with dedicated ui?

### DISCUSSION THREAD

**Discussion by hpruszyn** [vendurehq/vendure#1568] [Q&A] [answered]
Hi,
I'm investigating your product in context to use it as PIM (product Information Management) and later as full blown commerce engine. 
CPQ functionalities are clear and meet my expectations. However I'm missing PIM functionalities. Is it possible to create different product/variants definitions with different attributes? 

For example I would like to sell trees and flowers, both have attributes like "name", "description" but only flowers have attribute "color", and only trees have attributes "height": any number, "type" with possible values: conifer, deciduous tree. Is it possible to model it? Facets are not enough because when creating product I would like to choose what kind of product I would like to create and I would like to know what attributes (facets) are dedicated for this product. I would like to see only attributes that are dedicated for this product, e.g.: creating tree, I don't want to see color attribute.

**Comment by michaelbromley** [comment, marked as answer]
Hi,

What you describe is not natively supported by Vendure currently. The closest thing might be the use of [custom fields with tabs](https://www.vendure.io/docs/developer-guide/customizing-models/#tabbed-custom-fields), so e.g. you can have a tab for "flowers" containing the relevant custom field inputs, and then one for "trees" etc.

**Reply by hpruszyn** [reply]
Ok, thank you for response. 

I didn’t see this feature on the road map, is it possible to have it in next couple of months? Otherwise it is not enough to use Vendure in business with more complex product catalogs.

**Reply by michaelbromley** [reply]
That's right, it's not on the roadmap currently. It could be a candidate for v2 potentially, although even in that case, I wouldn't expect it to be available within 2 months. There are currently higher-priority features for v2 that I would work on first.

**Reply by hpruszyn** [reply]
Understood. I'll keep an eye on Vendure and maybe my next project can benefit from it. Thanks for discussion.
